### ISSUE TITLE
batch of prompt for stable diffusion model

### ISSUE THREAD

**Issue by Suzan009** [keras-team/keras-cv#1449]
Generally it does as follows

```
images = model.text_to_image(
    "a cute magical flying dog, fantasy art, "
    "golden color, high quality, highly detailed, elegant, sharp focus, "
    "concept art, character concepts, digital painting, mystery, adventure",
    batch_size=3,
)
```

Now, how to extend this for batch of prompt? To run the stable diffusion on TPU,  francois showed and shared a notebook https://colab.research.google.com/drive/17zQOm_2Iu6pcP8otT-v6rx0D-pKgfaLm?usp=sharing, in that colab, it is assumed that batch size = 8 will be braodcast to all 8 core of TPU, but if I want to run 8 promot with 8 image each prompt, how should I change the code?

**Comment by lb-97** [comment]
Hi @Suzan009, 
This [website](https://keras.io/guides/keras_cv/generate_images_with_stable_diffusion/#introduction) by Keras gives various examples on how you can pass a set of prompts. Under Introduction section 2nd example, you can see that there were three images produced for three different prompts. So, the code you posted above should work. Hope this resolves your query.

**Comment by Suzan009** [comment]
You mean this 

```
images = model.text_to_image(
    "a cute magical flying dog, fantasy art, "
    "golden color, high quality, highly detailed, elegant, sharp focus, "
    "concept art, character concepts, digital painting, mystery, adventure",
    batch_size=3,
)
``

If so, I think you're mistaken, that is one prompt, written in multiple rows.

**Comment by lb-97** [comment]
@Suzan009, yes you're right. 
On that note, could you not achieve what you want by iteratively making function calls? I believe this is a simpler approach. Or I can raise a PR to accommodate your request setting - batchify prompts.

**Comment by Suzan009** [comment]
> On that note, could you not achieve what you want by iteratively making function calls? I believe this is a simpler approach.

Yes, that is possible. Now, when I tried to run stable diffusion on tpu with one prompt and with one batch (as my per requirements), I felt like why not passing 8 prompt to 8 core of TPU to get best of it. From the above colab notebook, it is written as 

```
imgs = generator.generate(
    "Ruins of a castle in Scotland",
    num_steps=50,
    unconditional_guidance_scale=7.5,
    temperature=1,
    batch_size=8,
)
```
here one prompt generates 8 images on 8 core of TPU. If I set batch size to 1, the usage of tpu is super slow and inefficient. Most of the core of tpu became sets as idle. 

Honestly, I'm bit confuse to how to formulate the issue here. If I set (let's say for tpu), 1 prompt + 8 images, can I extend it as 8 prompt to 8 images or 8 prompt to 8 * 8 images? Now, afaik, stable diffusion by its core work on single prompt, in that case, that might be an issue as I need to vectorize the generative ops in order to batchify the prompt. Or, I may use tf.data API for the prompts to efficiently parse towards the model.
